snapcraft/_store.py

```diff
@@ -73,6 +75,84 @@ def logout():
     logger.info('Credentials cleared.')
 
 
+def _get_usable_secret_keys(query=None):
+    ctx = gpgme.Context()
+    for key in ctx.keylist(query, True):
+        if not key.subkeys or not key.uids or key.disabled or not key.can_sign:
+            continue
+        mainkey = key.subkeys[0]
+        if (mainkey.expired or mainkey.revoked or mainkey.disabled or
+                mainkey.invalid):
+            continue
+        # We're currently only interested in RSA >= 4096-bit.
+        if mainkey.pubkey_algo != gpgme.PK_RSA or mainkey.length < 4096:
+            continue
+        yield key
+
+
+def _display_fingerprint(fingerprint):
+    assert len(fingerprint) == 40
```

Comment:
I don't like assertions. I think it's better to make an if and raise the exception explicitly. The "assert" statement can be disabled and the "if" can't, but that's not likely to happen. My main problem with assertions is that they are intended to discover errors from the programmer, things that should never happen. And for me, that's what unit tests are for. So if you already have a test that makes sure this condition can't happen, the statement is not useful. And if this can happen during the normal operation of the program, that's what exceptions are for, and they are clearer.